tests/decimal_tests.rs

```diff
@@ -3397,8 +3399,15 @@ mod maths {
             let x = Decimal::from_str(x).unwrap();
             let tolerance = Decimal::from_str(tolerance).unwrap();
             let expected = Decimal::from_str(expected).unwrap();
-            assert_eq!(expected, x.exp_with_tolerance(tolerance));
+            assert_eq!(expected, x.exp_with_tolerance(tolerance).unwrap());
         }
+
+        let tolerance = Decimal::from_str("0.0002").unwrap();
```

Comment:
It'd be nice to have another set of tests similar to this:

https://github.com/paupino/rust-decimal/blob/a729c0d0c6d1d73432128d2cb8c5eb6e14f2a194/tests/decimal_tests.rs#L3347-L3378

The difference would be, of course, that they call `checked_exp`. It doesn't need to be quite as exhaustive, but would be nice to return one with a `Some` value and one with a known `None` value. Something like (untested):

```rust
#[cfg(not(feature = "legacy-ops"))]
#[test]
fn test_checked_exp() {
    let test_cases = &[
        // e^10 ~= 22026.465794806703
        ("10", Some("22026.416157416030662013737698")),
        // I'm not sure if this fails, it's just an educated guess
        ("1234567891234", None),
    ];

    for &(x, expected) in test_cases {
        let x = Decimal::from_str(x).unwrap();
        let result = x.checked_exp();
        if let Some(expected) = expected {
            assert!(result.is_some(), "Expected result to be Some but was None");
            assert_eq!(result.unwrap(), Decimal::from_str(expected).unwrap());
        } else {
            assert!(result.is_none(), "Expected result to be None but was Some: {}", result);
        }
}
```